package.json

```diff
@@ -74,6 +73,11 @@
   "engines": {
     "node": ">=8.0"
   },
+  "husky": {
+    "hooks": {
+      "pre-commit": "npm run npmcheck"
+    }
```

Comment:
> on the other hand we might be able to just use a pre-commit hook for the npm-shriwnkwrap.json and run this check in there. We do have a similar hook in the UI for deduplicating our yarn.lock. What do you think?
See: https://github.com/balena-io/resin-ui/blob/aa6e6a8c56d85281766b4f4715d30b7efedd154b/.lintstagedrc#L12

@thgreasi, I think it's a good idea. I understand that `resin-ui` and `resin-api` use `husky` for this. I've added husky above (using `package.json` rather than `.huskyrc` for its configuration). I think the check could be added in addition to `prebuild`, because git hooks are also not guaranteed to run on the client side. Belts and braces..

Reply:
Ha. There's a complication with `prebuild` as well. The balena CI build was failing on Windows because it uses Node 8 and runs `npx npm@6.9.0 install` instead of `npm install` -- it was the CI's workaround to avoid the npm bug. This means that a `preinstall` check for v6.9.0 passes, but a `prebuild` check fails as the default npm is v6.4.1. It's a to-do item to update balena CI on Windows to use Node 10 and/or Node 12...
So we are left with fewer choices for this PR, and the husky pre-commit hook may be the best one, at least for now.

Reply:
@pdcastro I generally prefer `.huskyrc` since it keeps the package.json cleaner and we can easily omit unwanted files from the published ones (eg in the SDK). Plus sounds nearer to the single responsibility principle. That's of course just imo and just a NIT, but it's great anyway that you added that hook as part of this PR :+1: